File: hw/bonilla/U3/HW23Strategy/Strategy/src/main/java/ec/edu/espe/strategy/controller/InsertionSort.java
package ec.edu.espe.strategy.controller;

/**
 *
 * @author Bonilla David SoftCrafters
 */
public class InsertionSort implements SortingStrategy {

    @Override
    public int[] sort(int data[]) {
        System.out.println("Usando Insertion Sort...");
        int[] sortedData = data.clone();

        for (int i = 1; i < sortedData.length; i++) {
            int key = sortedData[i];
            int j = i - 1;

            while (j >= 0 && sortedData[j] > key) {
                sortedData[j + 1] = sortedData[j];
                j = j - 1;
            }
            sortedData[j + 1] = key;
        }

        mostrarArreglo(sortedData);
        return sortedData;
    }

    private void mostrarArreglo(int[] data) {
        System.out.print("Resultado: ");
        for (int num : data) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}

File: ws/bonilla/U3/WS34Strategy/Strategyjs/controller/InsertionSort.js
const SortingStrategy = require('./SortingStrategy');

class InsertionSort extends SortingStrategy {
    sort(data) {
        console.log("Usando Insertion Sort...");
        const sortedData = [...data];

        for (let i = 1; i < sortedData.length; i++) {
            const key = sortedData[i];
            let j = i - 1;

            while (j >= 0 && sortedData[j] > key) {
                sortedData[j + 1] = sortedData[j];
                j--;
            }
            sortedData[j + 1] = key;
        }

        this.mostrarArreglo(sortedData);
        return sortedData;
    }

    mostrarArreglo(data) {
        process.stdout.write("Resultado: ");
        data.forEach(num => process.stdout.write(num + " "));
        console.log();
    }
}

module.exports = InsertionSort;
